app/services/ReservationService.java

```diff
@@ -0,0 +1,191 @@
+                }
+            }
+        }
+
+        Long timeToCheckUp = dateAndTime;
+        Long timeToCheckDown = dateAndTime;
+
+        int loopBreaker = 0;
+
+        while (bestTime.size() < 4) {
+            loopBreaker++;
+
+            if (timeToCheckUp != null) {
+                if (hasFreeTables(reservations, timeToCheckUp)) {
+                    Date date2 = new Date(timeToCheckUp);
+                    DateFormat formatter = new SimpleDateFormat("HH:mm");
+                    bestTime.add(formatter.format(date2));
+
+                }
+            }
+
+            timeToCheckUp = stepUp(timeToCheckUp);
+            timeToCheckDown = stepDown(timeToCheckDown);
+
+            if (timeToCheckDown != null) {
+                if (hasFreeTables(reservations, timeToCheckDown)) {
+                    Date date2 = new Date(timeToCheckDown);
+                    DateFormat formatter = new SimpleDateFormat("HH:mm");
+                    bestTime.add(formatter.format(date2));
+
+                }
+            }
+
+            if(loopBreaker > 30){
+                break;
+
+            }
+        }
+
+        return bestTime;
+
+    }
+
+
+    private Long stepUp(Long time) {
+        if (time != null) {
+            if (time <= LAST_RESERVATION) {
+                return time + HALF_HOUR;
+            }
+        }
+        return null;
+
+    }
+
+
+    private Long stepDown(Long time) {
+        if(time != null) {
+            if (time > FIRST_RESERVATION) {
+                return time - HALF_HOUR;
+            }
+        }
+        return null;
+    }
+
+
+    private List getFirstReservation(Long timestamp) {
+        DateFormat formatter = new SimpleDateFormat("HH:mm");
+        ArrayList<String> bestTimes = new ArrayList<>();
+
+        bestTimes.add(formatter.format(new Date(timestamp - HALF_HOUR)));
+        bestTimes.add(formatter.format(new Date(timestamp)));
+        bestTimes.add(formatter.format(new Date(timestamp + HALF_HOUR)));
+        bestTimes.add(formatter.format(new Date(timestamp + HOUR)));
+
+        return bestTimes;
+    }
+
+    public ReservationResponse getReservation(Restaurant restaurant, Reservation reservation) {
+
+        Long restaurantId = restaurant.getId();
+        String restaurantName = restaurant.getName();
+        String restaurantImage = restaurant.getImage();
+        int people = reservation.getNumberOfPeople();
+        int numOfFreeTables = getTablesForSelectedNumberOfPeople(restaurantId, people).size();
+        String date = reservation.getDate();
+        String time = reservation.getTime();
+        Timestamp timestamp = Timestamp.valueOf(date + " " + time + ":00.0");
+        Long timestamp2 = timestamp.getTime();
+
+        List<String> bestTimes = getBestTimes(date, timestamp2, restaurantId, people);
+
+        ReservationResponse reservationResponse = new ReservationResponse(numOfFreeTables, bestTimes, restaurantId, restaurantName, restaurantImage, timestamp, people, time);
```

Comment:
Can you just pass `restaurant` in this constructor instead of `id`, `name` and `image` so there is less arguments.